src/rules/objectLiteralSortKeysRule.ts

```diff
@@ -52,17 +54,23 @@ export class Rule extends Lint.Rules.OptionallyTypedRule {
                 const obj: I = { foo: 1, bar: 2 };
 
             If a contextual type is not found, alphabetical ordering will be used instead.
+            * "${OPTION_MATCH_DECLARATION_ORDER_ONLY}" exactly like "${OPTION_MATCH_DECLARATION_ORDER}",
+                but don't fall back to alphabetical if a contextual type is not found.
+
+                Note: If both ${OPTION_MATCH_DECLARATION_ORDER_ONLY} and ${OPTION_MATCH_DECLARATION_ORDER} options are present,
+                      ${OPTION_MATCH_DECLARATION_ORDER_ONLY} will take precedence and alphabetical fallback will not occur.
+
             * "${OPTION_SHORTHAND_FIRST}" will enforce shorthand properties to appear first, as in:
 
                 const obj = { a, c, b: true };
             `,
         options: {
             type: "string",
-            enum: [OPTION_IGNORE_CASE, OPTION_MATCH_DECLARATION_ORDER, OPTION_SHORTHAND_FIRST],
+            enum: [OPTION_IGNORE_CASE, OPTION_MATCH_DECLARATION_ORDER, OPTION_MATCH_DECLARATION_ORDER_ONLY, OPTION_SHORTHAND_FIRST],
         },
         optionExamples: [
             true,
-            [true, OPTION_IGNORE_CASE, OPTION_MATCH_DECLARATION_ORDER, OPTION_SHORTHAND_FIRST],
+            [true, OPTION_IGNORE_CASE, OPTION_MATCH_DECLARATION_ORDER, OPTION_MATCH_DECLARATION_ORDER_ONLY, OPTION_SHORTHAND_FIRST],
```

Comment:
add a new option example with this new option and _not_ `OPTION_MATCH_DECLARATION_ORDER`. let's not encourage users to set both.